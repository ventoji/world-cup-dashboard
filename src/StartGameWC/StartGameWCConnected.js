import React from 'react';
import { connect } from 'react-redux';
import StartGameWC from './StartGameWC';
import { addMatchWC } from '../store/actions/actions';
import { nameCountries } from '../utils';

export const StartGameWCConnected = ({ countries, addMatchWC }) => {
  const onSubmit = (mathDetails) => {
    addMatchWC(mathDetails);
  };
  return <StartGameWC onSubmit={onSubmit} countries={countries} />;
};

const mapStatetoProps = (state) => {
  return {
    countries: nameCountries(state.countries),
  };
};
const mapDispatchToProps = (dispatch) => ({
  addMatchWC: (mathDetails) => dispatch(addMatchWC(mathDetails)),
});

export default connect(
  mapStatetoProps,
  mapDispatchToProps,
)(StartGameWCConnected);
