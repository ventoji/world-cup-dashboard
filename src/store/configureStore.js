import { createStore, combineReducers } from 'redux';
import reducerActiveGames from './reducer';
import reducerSumaryResult from './reducerSummaryResult';

const rootReducer = combineReducers({
  activeGames: reducerActiveGames,
  finishedGames: reducerSumaryResult,
});

// const composeEnhancers = (typeof window !== 'undefined' && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;

export default () => {
  const store = createStore(
    rootReducer,
    window.__REDUX_DEVTOOLS_EXTENSION__ &&
      window.__REDUX_DEVTOOLS_EXTENSION__(),
  );

  return store;
};
