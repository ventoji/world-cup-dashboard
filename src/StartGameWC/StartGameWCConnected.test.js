import React from 'react';
import { Provider } from 'react-redux';
// import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';

import { StartGameWCConnected } from './StartGameWCConnected';

const mockStore = configureStore([]);

describe('StartGameWCConnected component', () => {
  let store;
  let component;
  let props;

  beforeEach(() => {
    store = mockStore({
      state: {
        countries: [],
        startedGames: [],
        summaryResult: [],
      },
    });
    store.dispatch = jest.fn();

    props = {
      addMatchWC: jest.fn(),
      countries: [],
    };
    component = shallow(
      <Provider store={store}>
        <StartGameWCConnected {...props} />
      </Provider>,
    ).dive();
  });

  it('Should be defined', () => {
    expect(component).toBeDefined();
  });

  it('should render with given state from Redux store', () => {
    expect(toJson(component)).toMatchSnapshot();
    expect(component.find('StartGameWCConnected')).toHaveLength(1);
    //    expect(component.find('Connect(StartGameWCConnected)')).toHaveLength(1);
    // expect(store.dispatch).toHaveBeenCalledTimes(1);
  });
});
