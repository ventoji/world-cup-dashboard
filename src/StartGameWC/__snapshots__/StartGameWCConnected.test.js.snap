// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartGameWCConnected component should render with given state from Redux store 1`] = `
<StartGameWCConnected
  addMatchWC={[MockFunction]}
  countries={Array []}
/>
`;
